openmc/model/model.py

```diff
@@ -151,44 +206,156 @@ def from_xml(cls, geometry='geometry.xml', materials='materials.xml',
         settings = openmc.Settings.from_xml(settings)
         return cls(geometry, materials, settings)
 
-    def deplete(self, timesteps, chain_file=None, method='cecm',
-                fission_q=None, **kwargs):
+    def init_lib(self, threads=None, geometry_debug=False, restart_file=None,
+                 tracks=False, output=True, event_based=None, intracomm=None):
+        """Initializes the model in memory via the C API
+
+        .. versionadded:: 0.13.0
+
+        Parameters
+        ----------
+        threads : int, optional
+            Number of OpenMP threads. If OpenMC is compiled with OpenMP
+            threading enabled, the default is implementation-dependent but is
+            usually equal to the number of hardware threads available
+            (or a value set by the :envvar:`OMP_NUM_THREADS` environment
+            variable).
+        geometry_debug : bool, optional
+            Turn on geometry debugging during simulation. Defaults to False.
+        restart_file : str, optional
+            Path to restart file to use
+        tracks : bool, optional
+            Write tracks for all particles. Defaults to False.
+        output : bool
+            Capture OpenMC output from standard out
+        event_based : None or bool, optional
+            Turns on event-based parallelism if True. If None, the value in
+            the Settings will be used.
+        intracomm : DummyCommnicator, mpi4py.MPI.Intracomm or None, optional
+            MPI intracommunicator
+        """
+
+        # TODO: right now the only way to set most of the above parameters via
+        # the C API are at initialization time despite use-cases existing to
+        # set them for individual runs. For now this functionality is exposed
+        # where it exists (here in init), but in the future the functionality
+        # should be exposed so that it can be accessed via model.run(...)
+
+        args = _process_CLI_arguments(
+            volume=False, geometry_debug=geometry_debug,
+            restart_file=restart_file, threads=threads, tracks=tracks,
+            event_based=event_based)
+        # Args adds the openmc_exec command in the first entry; remove it
+        args = args[1:]
+
+        self.finalize_lib()
+
+        # The Model object needs to be aware of the communicator so it can
+        # use it in certain cases, therefore lets store the communicator
+        if intracomm is not None:
+            self._intracomm = intracomm
+        else:
+            self._intracomm = DummyCommunicator()
+
+        if self._intracomm.rank == 0:
+            self.export_to_xml()
+        self._intracomm.barrier()
+
+        openmc.lib.init(args=args, intracomm=self._intracomm, output=output)
```

Comment:
Right now this doesn't work if you pass a `DummyCommunicator` but `mpi4py` is installed. I found this out when I tried to run tests on my machine using your branch and were getting failures. An easy fix is:
```suggestion
        openmc.lib.init(args=args, intracomm=intracomm, output=output)
```